### DISCUSSION TITLE
Is it able to extract certain sub tree after calling TransformToTree?

### DISCUSSION THREAD

**Discussion by Snailya** [reactivemarbles/DynamicData#882] [Q&A] [answered]
hi guys,
I want to filter a tree to get only the subtree that meets the certain id. I've tried to call Filter() after TransformToTree, but it returns nothing.
I currently came up a reluctant way that is firstly filter out items that belong to that subtree then call TransformToTree. But I wonder if there's other amazing Dynamic way to do it?

**Comment by Snailya** [comment, marked as answer]
I've figured out the correct way is to create a predicate function for TransformToTree.
The default predicate for the TransformToTree example is
```
Observable.Return((Node<EmployeeDto, int> node) => node.IsRoot);
```
For my situation, is to build a predicate based on the id of the node.
```
var categoryPredicate = this.WhenAnyValue(x => x.ElementName)
                            .Select((string name) => Node<EmployeeDto, int> node) => service.GetValidIds(name).Contains(node.Id));
```

**Comment by JakenVeina** [comment]
Sorry if it seemed like you were being ignored. I know absolutely nothing about the tree APIs myself, so I figured I'd let others answer, and then I forgot about this post. Otherwise, I'd have done some reasearch.

Glad you've got it figured out, thanks for the answer.

**Reply by Snailya** [reply]
Never mind.
The dynamic data is amazing. I'm really grateful that you guys paying time to maintain and imporve it.
